syft/frameworks/torch/hook/hook.py

```diff
@@ -504,6 +510,104 @@ def overloaded_attr(self, *args, **kwargs):
 
         return overloaded_attr
 
+    def _hook_promise_tensor(hook_self):
```

Comment:
Why do we need to hook the promise tensor? Why can't we just define all the methods directly in the PromiseTensor class? The class resides within PySyft, so why don't we define the methods as for example `DoubleTensor` there? It doesn't seem to rely on the DoubleTensor being hooked before defining the method.

Reply:
Methods like `DoubleTensor` were in the file where the class is defined before but I was asked in some comments to move them here ^^
For the other methods, I think this file was supposed to be where this kind of method generation happen but maybe not?

Reply:
Ok, that sounds like a contradiction. I'd like to hear the opinion of @LaRiffle and @robert-wagner :smile:

Reply:
Historically PromiseTensor has always been a little bit of an exception because of the way it works. I'm ok with this for now.